scripts/code_style.py

```diff
@@ -63,31 +63,41 @@ def list_generated_files() -> FrozenSet[str]:
     checks = re.findall(CHECK_CALL_RE, content)
     return frozenset(word for s in checks for word in s.split())
 
-def get_src_files() -> List[str]:
+def get_src_files(since: Optional[str]) -> List[str]:
     """
-    Use git ls-files to get a list of the source files
-    """
-    git_ls_files_cmd = ["git", "ls-files",
-                        "*.[hc]",
-                        "tests/suites/*.function",
-                        "scripts/data_files/*.fmt"]
+    Use git to get a list of the source files.
 
-    result = subprocess.run(git_ls_files_cmd, stdout=subprocess.PIPE,
-                            check=False)
+    The optional argument since is a commit, indicating to only list files
+    that have changed since that commit. Without this argument, list all
+    files known to git.
 
-    if result.returncode != 0:
-        print_err("git ls-files returned: " + str(result.returncode))
-        return []
+    Only C files are included, and certain files (generated, or 3rdparty)
+    are excluded.
+    """
+    if since is None:
+        git_ls_files_cmd = ["git", "ls-files",
+                            "*.[hc]",
+                            "tests/suites/*.function",
+                            "scripts/data_files/*.fmt"]
+        output = subprocess.check_output(git_ls_files_cmd,
+                                         universal_newlines=True)
     else:
-        generated_files = list_generated_files()
-        src_files = str(result.stdout, "utf-8").split()
-        # Don't correct style for third-party files (and, for simplicity,
-        # companion files in the same subtree), or for automatically
-        # generated files (we're correcting the templates instead).
-        src_files = [filename for filename in src_files
-                     if not (filename.startswith("3rdparty/") or
-                             filename in generated_files)]
-        return src_files
+        git_ls_files_cmd = ["git", "diff", "--name-only", since, "--",
```

Comment:
I think the files found by `git diff --name-only since --` should be a subset of `git ls-files`. Because `git diff` might list some files which are not present in `mbedtls` anymore.
For example:
```
$ ./scripts/code_style.py -s HEAD~10
Error:  Uncrustify returned 1 correcting file library/x509_invasive.h
```
As we have removed `library/x509_invasive.h` by #7638